include/swift/AST/Decl.h

```diff
@@ -1098,11 +1098,22 @@ class alignas(1 << DeclAlignInBits) Decl : public ASTAllocated<Decl> {
 
   bool isAvailableAsSPI() const;
 
-  /// Whether the declaration is considered unavailable through either being
-  /// explicitly marked as such, or has a parent decl that is semantically
-  /// unavailable. This is a broader notion of unavailability than is checked by
-  /// \c AvailableAttr::isUnavailable.
-  bool isSemanticallyUnavailable() const;
+  /// Retrieve the @available attribute that provides the OS version range that
+  /// this declaration is available in.
+  ///
+  /// The attribute may come from another declaration, since availability
+  /// could be inherited from a parent declaration.
```

Comment:
Nit: Could you please comment on the meaning of the `Decl` returned in the pair? 
I gather that it's the decl whose availability ultimately was used (either the current decl itself or the one it inherits from or the decl of the nominal this one extends, from what I gather).

Reply:
That's a good idea; if you don't mind I'm going to do so in an upcoming PR where I adopt these new utilities elsewhere, just to avoid another full CI cycle.

Reply:
Yes, of course. 👍🏼